integration-tests/gradle/build.gradle.kts

```diff
@@ -0,0 +1,124 @@
+val expectedReducedRuntimeClasspathJava6 = setOf(
+        "guava-HEAD-android-SNAPSHOT.jar",
+        "failureaccess-1.0.1.jar"
+)
+val expectedReducedRuntimeClasspathJava8 = setOf(
+        "guava-HEAD-jre-SNAPSHOT.jar",
+        "failureaccess-1.0.1.jar"
+)
+val expectedCompileClasspathJava6 = expectedReducedRuntimeClasspathJava6 + setOf(
+        "jsr305-3.0.2.jar",
+        "checker-compat-qual-2.5.5.jar",
+        "error_prone_annotations-2.3.4.jar",
+        "j2objc-annotations-1.3.jar"
+)
+val expectedCompileClasspathJava8 = expectedReducedRuntimeClasspathJava8 + setOf(
+        "jsr305-3.0.2.jar",
+        "checker-qual-2.11.1.jar",
+        "error_prone_annotations-2.3.4.jar",
+        "j2objc-annotations-1.3.jar"
+)
+
+val extraLegacyDependencies = setOf(
+        "listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar",
+        "google-collections-1.0.jar"
+)
+
+subprojects {
+    apply(plugin = "java-library")
+
+    val expectedClasspath =
+        if (gradle.gradleVersion.startsWith("5.")) {
+            // without Gradle Module Metadata (only the POM is used)
+            // - variant decission is made based on version suffix (android/jre) and not on actual Java version
+            // - runtime classpath equals the compile classpath
+            // - dependency conflict with Google Collections is not detected and '9999.0' hack is present
+            if (name.startsWith("android")) {
+                expectedCompileClasspathJava6 + extraLegacyDependencies
+            } else {
+                expectedCompileClasspathJava8 + extraLegacyDependencies
+            }
+        } else {
+            // with Gradle Module Metadata
+            // - variant is choosend based on Java version used independent of version suffix
+            // - reduced runtime classpath is used (w/o annotation libraries)
+            // - capability conflicts are detected between Google Collections and Listenablefuture
+            if (name.contains("Java6")) {
+                if (name.contains("runtimeClasspath")) {
+                    expectedReducedRuntimeClasspathJava6
+                } else {
+                    expectedCompileClasspathJava6
+                }
+            } else {
+                if (name.contains("runtimeClasspath")) {
+                    expectedReducedRuntimeClasspathJava8
+                } else {
+                    expectedCompileClasspathJava8
+                }
+            }
+        }
+    val guavaVersion = if (name.startsWith("jre")) {
+        "HEAD-jre-SNAPSHOT"
```

Comment:
It sounds like we may need to update this manually (or tweak `set_version` to update it with `sed` or something) to test against a version that we're about to release? (And similarly at the top of the file.)

That should be fine. I just want to check my understanding (and invite you to let me know if there are other things we should do that I've overlooked).

Reply:
You are right. I missed this.

I fixed it by extracting the version directly from the `pom.xml` file.